common/client/models.go

```diff
@@ -74,3 +75,45 @@ func (n NodeTier) String() string {
 		return fmt.Sprintf("NodeTier(%d)", n)
 	}
 }
+
+// syncIssue - defines problems related to RPC's state synchronization. Can be used as a bitmask to define multiple issues
+type syncIssue int
+
+const (
+	// syncIssueNotInSyncWithPool - RPC is lagging behind the highest block observed within the pool of RPCs
+	syncIssueNotInSyncWithPool syncIssue = 1 << iota
+	// syncIssueNoNewHead - RPC failed to produce a new head for too long
+	syncIssueNoNewHead
+	// syncIssueNoNewFinalizedHead - RPC failed to produce a new finalized head for too long
+	syncIssueNoNewFinalizedHead
+	syncIssueLen
+)
```

Comment:
Does changing `syncIssue` to `syncStatus` and adding `synced` state as 0 work, or does it mess up the enumeration incrementation? I noticed in `outOfSyncLoop` we use a 0 value instead of an enum value because of this.